guides/Overmap Ships and Places - What's Needed.md

```diff
@@ -0,0 +1,59 @@
+  icon_state = "shuttlered"
+  requires_power = 1 // Or 0 if you don't want to have to power the ship yourself. 
+//Area should probably be placed in southern-cross-areas.dm but the rest can go into some file all together.
+  
+/datum/shuttle/autodock/overmap/SHIPNAME
+    name = "[Name]"
+    warmup_time = 0
+    current_location = "" // See CREATED IN-EDITOR
+    shuttle_area = /area/shuttle/SHIPNAME // See above area. Can also be list of multiple areas.
+    fuel_consumption = 1 // Typically defines how much fuel the shuttle uses per jump. 1-3 is reasonable, base it on size.
+    move_direction = NORTH / SOUTH / EAST / WEST
+
+/obj/effect/overmap/visitable/ship/landable/SHIPNAME
+    name = "[Name]" // Can be named whatever, best to keep it similar to actual ship name.
+    desc = "Some description goes here."
+    vessel_mass = 1000 // Some factor for speed. For reference, 1000 is fucking fast, 10000 is only fast when maxed on engines and very costly.
+    vessel_size = SHIP_SIZE_TINY / SHIP_SIZE_SMALL / SHIP_SIZE_LARGE
+    shuttle = "[Name]" // Direct reference to above shuttle. MUST be the same as name.
+
+/obj/machinery/computer/shuttle_control/explore/SHIPNAME
+    name = "short jump console"
+    shuttle_tag = "[Name]" // Direct reference to above shuttle. MUST be the same as name.
+    req_one_access = list(access_pilot)
+```
+
+**OVERMAP SHIP CREATION INSTRUCTIONS:**
+Build your shuttle however you'd like, though it MUST have:
+- Phoron cannister(s) hooked up to proper engines. /obj/machinery/atmospherics/unary/engine
+- Engine Control. `/obj/machinery/computer/ship/engines`
+- Helm Control. `/obj/machinery/computer/ship/helm`
+- Fuel port, for short distance jumps and docking. `/obj/structure/fuel_port`
+- Short jump console, defined in code earlier. `/obj/machinery/computer/shuttle_control/explore/SHIPNAME`
+- A spawned in effect, defined in code earlier, inside of the ship area. `/obj/effect/overmap/visitable/ship/landable/SHIPNAME`
+- A docking point, made through the map editor, mentioned earlier. `/obj/effect/ship_landmark`. Typically this is placed inside the ship, directly in front of the exterior door. This dictates where it places the shuttles so be sure to actually test how your shuttles react to landing in odd locations. The landmark's tag also needs to be added to the appropriate area in `maps/southern_cross/overmap/sectors.dm` or else ships cannot dock there.
+- The shuttle also needs to be placed entirely in the designated area that you made in the above code, since shuttles are based off of the area everything is in.
+
+It would be **RECOMMENDED** to also have:
+- Sensors, controlled by `/obj/machinery/computer/ship/sensors` and `/obj/machinery/shipsensors` (maybe the /weak variant for small ships if you add them at all?)
+- Docking Controller. Not 100% sure on how these work yet, but will be updated as needed. Not necessary, just nice to have for ease of access.
```

Comment:
Docking controllers are added to the `/datum/shuttle/autodock/overmap/SHIPNAME` *e.g.* `docking_controller_tag = "SHIPNAME_docking"` and I *believe* if you've set up the airlock properly then it should automatically link up with the docking port controller and helm so that both sides of an airlock can fully open if atmospheres on all sides are equivalent and the correct docking code is provided.

On that note you should probably include instructions on setting up the airlock and using Aro's airlock helpers, if not in this version then in the full guide!

Reply:
Probably will do in the full guide, yeah!
I just didn't have the energy to entirely write a step-by-step guide for this YET as I did with the new POI one.
Thank you!